.github/workflows/docker-publish.yml

```diff
@@ -40,5 +40,10 @@ jobs:
             azul_docker_pycharm_version=${{ env.azul_docker_pycharm_version }}
           context: .
           platforms: linux/amd64,linux/arm64
-          push: ${{ github.event_name != 'pull_request' && steps.vars.outputs.branch != 'master' }}
+          push: >
+            ${{
+            github.event_name != 'pull_request'
+            && steps.vars.outputs.branch != 'master'
+            && endsWith(steps.vars.outputs.branch, '-PR') != true
```

Comment:
`!` is a supported operator

Reply:
I had avoided the `!` operator due to [this thread](https://github.com/orgs/community/discussions/25017) claiming it gave a syntax error, however I did not confirm this. I will test.

Reply:
And you need to document quirks like that.